tests/conf/test_list_option.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import unittest
+
+import dnf
+
+
+class ListOptionTest(unittest.TestCase):
+
+    def setUp(self):
+        self.conf = dnf.conf.MainConf()
+
+    def test_iadd(self):
+        self.conf.pluginpath = ["a"]
```

Comment:
Is pluginpath a VectorString / ListOptionValue? If not, then these tests aren't testing the change. I'd rather see it use tsflags since that's what lorax uses and will be the first thing to explode :)

Reply:
It is, yeah. You can find the Sekrit Master List of options and their types [here](https://github.com/rpm-software-management/libdnf/blob/master/libdnf/conf/ConfigMain.cpp#L514-L612). `OptionStringList`s are the ones that behave somewhat normally, `OptionStringListAppend`s are the weird append-y ones.

Reply:
well, the weird behaviour of `OptionStringListAppend` is kind of orthogonal to this. I imagine @dmach would rather make any changes there separately. That odd 'append-by-default' behaviour was not actually introduced with the libdnf port - it was introduced before that, during the DNF 2.x series, with [this commit](https://github.com/rpm-software-management/dnf/commit/74ef3ff0922bce7511e9a9cdecce2cc921403d1b). So it's nothing to do with the problems of bridging C++ and Python, it's just a bad idea someone had. :P